include/envoy/stats/symbol_table.h

```diff
@@ -1,13 +1,157 @@
 #pragma once
 
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+#include "absl/strings/string_view.h"
+
 namespace Envoy {
 namespace Stats {
 
-// Interface for referencing a stat name.
 class StatName;
+class SymbolEncoding;
+
+/**
+ * Efficient byte-encoded storage of an array of tokens. The most common tokens
+ * are typically < 127, and are represented directly. tokens >= 128 spill into
+ * the next byte, allowing for tokens of arbitrary numeric value to be stored.
+ * As long as the most common tokens are low-valued, the representation is
```

Comment:
How is this guaranteed? I.e. how do you do the assignment of common stat names to low bytes? I'm sure I'll learn this if I read on, but I think the header should describe linearly enough background material to understand what is going on here.